win32ss/gdi/ntgdi/freetype.c

```diff
@@ -740,33 +740,16 @@ FtMatrixFromMx(FT_Matrix *pmat, PMATRIX pmx)
     pmat->yy = FLOATOBJ_GetLong(&ef);
 }
 
-VOID
+VOID FASTCALL
 FtSetCoordinateTransform(
     FT_Face face,
     PMATRIX pmx)
 {
-    FT_Matrix ftmatrix;
-    FLOATOBJ efTemp;
-
-    /* Create a freetype matrix, by converting to 16.16 fixpoint format */
-    efTemp = pmx->efM11;
-    FLOATOBJ_MulLong(&efTemp, 0x00010000);
-    ftmatrix.xx = FLOATOBJ_GetLong(&efTemp);
-
-    efTemp = pmx->efM12;
-    FLOATOBJ_MulLong(&efTemp, 0x00010000);
-    ftmatrix.xy = FLOATOBJ_GetLong(&efTemp);
-
-    efTemp = pmx->efM21;
-    FLOATOBJ_MulLong(&efTemp, 0x00010000);
-    ftmatrix.yx = FLOATOBJ_GetLong(&efTemp);
-
-    efTemp = pmx->efM22;
-    FLOATOBJ_MulLong(&efTemp, 0x00010000);
-    ftmatrix.yy = FLOATOBJ_GetLong(&efTemp);
+    FT_Matrix mat;
+    FtMatrixFromMx(&mat, pmx);
 
     /* Set the transformation matrix */
-    FT_Set_Transform(face, &ftmatrix, 0);
+    FT_Set_Transform(face, &mat, 0);
```

Comment:
Please note, that the FtMatrixFromMx() function https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/gdi/ntgdi/freetype.c;hb=774246e95c0df326464e59ad4a948c9dc8948d96#l721
places `pmx->efM21` into `ftmatrix.xy` (and vice-versa for efM12) instead of into `ftmatrix.xy` as was done originally there in FtSetCoordinateTransform().
My question therefore is, whether this is a bug or not ???

Reply:
Sorry, that was my mistake. Currently, `FtSetCoordinateTransform` is used in:
- `ftGdiGetGlyphOutline`
- `TextIntGetTextExtentPoint`
- `ftGdiGetTextMetricsW`
- `IntExtTextOutW`
- `NtGdiGetCharABCWidthsW`
- and `NtGdiGetCharWidthW`.

So at least we have to test `GetTextExtentPoint32W` with some `WorldTransform` matrixes.

Reply:
In case the difference would be intended, maybe add a comment to remember that.